### PR TITLE
refactor some stuff

### PR DESCRIPTION
**Pull request by Bjvanminnen**
- Extract getMaxExampleBlockWidth_
- Move some variables around so that they're closer to where we first use them
- Pass ExampleView a ContractEditor object instead of a couple of functions.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
Added another commit that moves onPlaceExampleContent into it's own function. It accessed a bunch of DOM elements that we're local variables, so I converted those to live as members of the ContractEditor.

File: blockly-core/core/ui/contract_editor/contract_editor.js [hunk 1]
```diff
@@ -872,3 +782,123 @@ Blockly.ContractEditor.prototype.changeParameterName_ = function(paramID, newNam
...
+/**
+ * Call our app-specific test handler for this block
+ */
+Blockly.ContractEditor.prototype.testExample = function (block) {
+  return this.testHandler_(block);
+};
+
+/**
+ * Call our app-specific test reset handler for this block
+ */
```
Comment (bcjordan) [inline blockly-core/core/ui/contract_editor/contract_editor.js]: add return jsdoc, e.g. `@return {string} result from reset`

File: blockly-core/core/ui/contract_editor/contract_editor.js [hunk 2]
```diff
@@ -872,3 +782,123 @@ Blockly.ContractEditor.prototype.changeParameterName_ = function(paramID, newNam
   var oldName = paramInfo.name;
   Blockly.Variables.renameVariable(oldName, newName, Blockly.modalBlockSpace)
 };
+
+/**
+ * Go through each of our example blocks and figure out which is widest
+ */
```
Comment (bcjordan) [inline blockly-core/core/ui/contract_editor/contract_editor.js]: `@return {number}` jsdoc

File: blockly-core/core/ui/contract_editor/contract_editor.js [hunk 3]
```diff
@@ -177,27 +177,27 @@ Blockly.ContractEditor.prototype.create_ = function() {
   this.exampleAreaDiv.style.position = 'absolute';
   goog.dom.insertChildAt(this.container_, this.exampleAreaDiv, 0);
 
-  var callText = goog.dom.createDom('div', 'callText');
-  callText.innerHTML = "Call";
-  goog.dom.appendChild(this.exampleAreaDiv, callText);
+  this.callText = goog.dom.createDom('div', 'callText');
```
Comment (bcjordan) [inline blockly-core/core/ui/contract_editor/contract_editor.js]: As you mentioned in HipChat, it does feel like we might want to pull this and other Example-section-specific state out into an e.g. `ExampleSectionView`, (maybe?) inheriting from `ContractEditorSectionView`, just to make the stored DOM element reference boundaries clearer.

A client of such an ExampleSectionView might want to then register `onRun` and `onReset` callbacks, so the ExampleSectionView doesn't need to know about the business logic of running examples, but just triggering them and displaying results (i.e., we should try to limit passed references to that).

File: blockly-core/core/ui/contract_editor/example_view.js [hunk 4]
```diff
@@ -52,14 +50,19 @@ Blockly.ExampleView.prototype.initializeTestButton = function (buttonText, iconC
 };
 
 Blockly.ExampleView.prototype.testExample_ = function () {
-  this.resultText.innerHTML = this.onExampleRun_(this.block_);
+  this.contractEditor_.resetExampleViews();
```
Comment (bcjordan) [inline blockly-core/core/ui/contract_editor/example_view.js]: Could the contract editor pass in references to resetExampleViews and testResetHandler_ as parameters? Just to clarify the boundary between the view and editor in the constructor signature.

**Comment by bcjordan** [comment]
Added comments, love the direction & thanks for early review!

File: blockly-core/core/ui/contract_editor/example_view.js [hunk 5]
```diff
@@ -52,14 +50,19 @@ Blockly.ExampleView.prototype.initializeTestButton = function (buttonText, iconC
...
   this.testRunning_ = true;
   this.refreshButtons();
+
   // TODO(bjordan): UI re-layout post-result?
 };
 
-Blockly.ExampleView.prototype.resetExample_ = function () {
-  this.onExampleReset_();
+Blockly.ExampleView.prototype.reset = function () {
+  this.contractEditor_.testResetHandler_();
```
Comment (Bjvanminnen) [inline blockly-core/core/ui/contract_editor/example_view.js]: I'm calling a private method. I shouldnt do that.

File: blockly-core/core/ui/contract_editor/example_view.js (hunk 4 above)
Comment (Bjvanminnen) [inline blockly-core/core/ui/contract_editor/example_view.js]: I'm not a huge fan of passing a bunch of callbacks to a constructor, as I end up finding this hard to follow.

Another approach we could take would be to make example_view an event emitter. When it emits "testing-example" the contract editor clears all example view and when it emits "reset" we would ask it for it's example block and reset that block.

Comment (bcjordan) [inline blockly-core/core/ui/contract_editor/example_view.js]: That could work! For emitters in closure-land, see e.g.

https://github.com/code-dot-org/code-dot-org/blob/vertical_scrollbars/blockly-core/core/ui/block_space/block_space.js#L79

or

https://github.com/code-dot-org/code-dot-org/blob/vertical_scrollbars/blockly-core/core/ui/block.js#L131-131

Comment (bcjordan) [inline blockly-core/core/ui/contract_editor/example_view.js]: Just need to take care to make sure we won't need to rely on execution order with multiple listeners down the road

File: blockly-core/core/ui/contract_editor/contract_editor.js (hunk 1 above)
Comment (Bjvanminnen) [inline blockly-core/core/ui/contract_editor/contract_editor.js]: Reset doesn't return anything as far as I can tell (only used by eval). However, I do want this for testExample which returns a string.
